test/integration/client/dynamic_client_test.go

```diff
@@ -203,6 +207,75 @@ func TestDynamicClientWatch(t *testing.T) {
 	}
 }
 
+func TestUnstructuredExtract(t *testing.T) {
+	result := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{"--disable-admission-plugins", "ServiceAccount"}, framework.SharedEtcd())
+	defer result.TearDownFn()
+
+	dynamicClient, err := dynamic.NewForConfig(result.ClientConfig)
+	if err != nil {
+		t.Fatalf("unexpected error creating dynamic client: %v", err)
+	}
+
+	resource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
+
+	// Apply an unstructured with the dynamic client
+	name := "test-pod"
+	pod := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "v1",
+			"kind":       "Pod",
+			"metadata": map[string]interface{}{
+				"name": name,
+				// namespace will always get set by extract,
+				// so we add it here (even though it's optional)
+				// to ensure what we apply equals what we extract.
+				"namespace": "default",
+			},
+			"spec": map[string]interface{}{
+				"containers": []interface{}{
+					map[string]interface{}{
+						"name":  "test",
+						"image": "test-image",
+					},
+				},
+			},
+		},
+	}
+	mgr := "testManager"
+	podData, err := json.Marshal(pod)
+	if err != nil {
+		t.Fatalf("failed to marshal pod into bytes: %v", err)
+	}
+
+	// apply the unstructured object to the cluster
+	actual, err := dynamicClient.Resource(resource).Namespace("default").Patch(
+		context.TODO(),
+		name,
+		types.ApplyPatchType,
+		podData,
+		metav1.PatchOptions{FieldManager: mgr})
+	if err != nil {
+		t.Fatalf("unexpected error when creating pod: %v", err)
+	}
+
+	// extract the object
+	discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(result.ClientConfig)
+	extractor, err := metav1ac.NewUnstructuredExtractor(discoveryClient)
+	if err != nil {
+		t.Fatalf("unexpected error when constructing extrator: %v", err)
+	}
+	extracted, err := extractor.Extract(actual, mgr)
```

Comment:
I think that's good, I wish maybe we could guarantee there are other fields. (to be fair, we know there are other fields because of defaulting, so that's also fine). We don't want to re-test the whole feature though. Thanks!

Reply:
I guess I could add an assertion for one of the defaulted fields (that it exists on actual, but not in extracted), but that seems a bit construed. 

I could also do a second apply with another manager, but like you said, that' more about testing the underlying `ExtractItems` rather than this specific unstructed one.

Either way let me know if either of these sound compelling or if you think of any way to validate the other fields, leaving as is for now.